src/Foreign/Haskell/Maybe.agda

```diff
@@ -1,38 +0,0 @@
-------------------------------------------------------------------------
--- The Agda standard library
---
--- Which Maybe type which calls out to Haskell via the FFI
-------------------------------------------------------------------------
-
-{-# OPTIONS --without-K #-}
-
-module Foreign.Haskell.Maybe where
-
-open import Level
-open import Data.Maybe.Base as Data using (just; nothing)
-
-private
-  variable
-    a : Level
-    A : Set a
-
-------------------------------------------------------------------------
--- Definition
-
-data Maybe (A : Set a) : Set a where
```

Comment:
For backwards compatibility we should really keep this module, but mark the internal terms as deprecated and have the warning explain they're no longer needed?